### PR TITLE
Add support for namespaces and respawn of crashed nodes

### PR DESCRIPTION
**Pull request by FlorianLebecque**
Added parameters in the launch file to set a namespace and node respawn.

```bash
ros2 launch rosbridge_server rosbridge_websocket_launch.xml namespace:="ROBOT" respawn:=true
```

This allow multiple robot to host their own bridge on the same network.
The respawn allows the Node to restart if it crashes. We had some issue with the ROSAPI node crashing when trying to access "dead" node information or node with empty names.

### CODE REVIEW

**Review by bjsowa — CHANGES_REQUESTED** [review]
The indentation after line 30 is broken. Please revert it to the current state.

**Review by bjsowa — CHANGES_REQUESTED** [review]
You changed the indentation from 2 spaces to 4. The only lines changed in this PR should be related to your feature

**Comment by FlorianLebecque** [comment]
Yes, indeed; should be fixed now

File: rosbridge_server/launch/rosbridge_websocket_launch.xml [hunk 1]
```diff
@@ -22,10 +24,14 @@
   <arg name="params_glob" default="" />
   <arg name="bson_only_mode" default="false" />
 
+  <arg name="respawn" default="false"/>
+
+  <arg unless="$(var bson_only_mode)" name="binary_encoder" default="default"/>
+
   <arg unless="$(var bson_only_mode)" name="binary_encoder" default="default"/>
```
Comment (bjsowa) [inline rosbridge_server/launch/rosbridge_websocket_launch.xml]: This line is doubled now. @sea-bass Is this even needed? This argument does not seem to be used anywhere

**Review by bjsowa — CHANGES_REQUESTED** [review]

File: rosbridge_server/launch/rosbridge_websocket_launch.xml (hunk 1 above)
Comment (FlorianLebecque) [reply rosbridge_server/launch/rosbridge_websocket_launch.xml]: I don't understand how i haven't see this.
The argument seems to be used in the rosbridge_websocket node

Comment (bjsowa) [reply rosbridge_server/launch/rosbridge_websocket_launch.xml]: The `bson_only_mode` is used but the `binary_encoder` defined here doesn't seem to be used

**Comment by bjsowa** [comment]
My proposed alternative for namespaces is to use:
```
  <push_ros_namespace namespace="$(var namespace)" />
```
before the nodes are defined.

File: rosbridge_server/launch/rosbridge_websocket_launch.xml [hunk 2]
```diff
@@ -6,6 +6,8 @@
   <arg name="certfile" default=""/>
   <arg name="keyfile" default="" />
 
+  <arg name="namespace" default="ws" />
```
Comment (bjsowa) [inline rosbridge_server/launch/rosbridge_websocket_launch.xml]: This should be empty by default to not change current behavior
```suggestion
  <arg name="namespace" default="" />
```

**Review by bjsowa — CHANGES_REQUESTED** [review]

**Review by sea-bass — APPROVED** [review]
Changes as they are look good and non disruptive.

Seems like everything was already resolved with @bjsowa 's review.

**Review by bjsowa — APPROVED** [review]
Seems fine to me. The only issue is that rosapi node [uses global namespace for all its services](https://github.com/RobotWebTools/rosbridge_suite/blob/1de68964d2cb0687284760ba45979ec2f584a5b9/rosapi/scripts/rosapi_node#L91-L130), so spawning multiple rosapi nodes in different namespaces creates a collision in service names.
roslibjs has a similar issue where it uses global namespace for service clients:
https://github.com/search?q=repo%3ARobotWebTools%2Froslibjs%20rosapi&type=code

Unless fixed in both projects, it won't work properly.

**Pull request merged**